EIPS/eip-5554.md

```diff
@@ -0,0 +1,213 @@
+requires: 5218
+---
+
+## Abstract
+
+This EIP extends any other token standard to provide:
+
+* Explicit rights for the token holder related to commercial exploitation, derivative works, and reproduction;
+* [EIP-5218](./eip-5218.md) interface for creating, viewing, and checking the status of licenses
+* Standard format for extended license information in the token metadata;
+* Standard events to track off chain creation of derivative works, commercial exploitation, and reproduction;
+* On chain tracking of derivative works and reproductions
+* Additional required fields in the smart contract to reference the copyright owner
+* Function calls for commercial exploitation, derivative works and reproduction.
+
+## Motivation
+NFTs still face legal uncertainty, and many now realize that the rights associated with an NFT are just as important as the NFT itself. Our goal is to help the ecosystem reach clear consensus and broad understanding of what purchasers of NFTs are acquiring in terms of copyright or other rights. 
+
+Today, purchasing the NFT of a digital work is not the same as purchasing the copyright in that work. In most cases, the NFT does not even incorporate the digital work; it only references it via a hash. Hence, the NFT holder owns a unique digital copy of the work, but does not necessarily enjoy the right to reproduce, redistribute, or otherwise exploit that work—unless explicitly provided for by the copyright owner. It typically only includes the right to privately enjoy the work and display it publicly on social media or in virtual galleries. 
+
+We aim to create a new set of licenses with modular terms and conditions—à la Creative Commons—in order to enable artists to increase the value of their NFT by associating additional rights to them (e.g. the right to create derivative works, or to allow for the commercial usage of the underlying works). Our solution will allow for any licensed rights to be granted, only and exclusively, to the current holders of an NFT, and to be transferred automatically to the new token holders every time the NFT is being transferred. 
+
+An on chain registry of copyrighted material will help in discovery of the rights associated with the NFTs that have been created with this protocol.
+
+Our current work is drafting the legalese and technical specifications.
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+Every contract compliant with this EIP must implement the `IERC5554` interface:
+
+```solidity
+pragma solidity ^0.8.0;
+
+interface IERC5554 is IERC5218 {
+
+    event CommercialExploitation(uint256 _tokenId, uint256 _licenseId, string _externalUri);
+    event ReproductionCreated(uint256 _tokenId, uint256 _licenseId, address _reproduction);
+    event DerivativeCreated(uint256 _tokenId, uint256 _licenseId, address _derivative);
```

Comment:
```suggestion
    event ReproductionCreated(uint256 _tokenId, uint256 _licenseId, uint256 _reproductionId, address _reproduction, uint256 _reproductionTokenId);
    event DerivativeCreated(uint256 _tokenId, uint256 _licenseId, uint256 _derivativeId, address _derivative, uint256 _derivativeTokenId);
```

1. If `_reproductionId` and `_derivativeId` is assigned at execution, it's better to log them in events as well for easy filtering and tracking;
2. The reproductions and derivatives may not exist as an individual collection but in collections of other works as well, e.g., consider the Opensea smart contract. So it's better to log the token ID for them in addition to the smart contract address.